frontend/src/features/leases/detail/LeasePages/payment/PaymentsContainer.test.tsx

```diff
@@ -0,0 +1,178 @@
+import userEvent from '@testing-library/user-event';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import { Claims } from 'constants/claims';
+import { LeaseStateContext } from 'features/leases/context/LeaseContext';
+import { Formik } from 'formik';
+import { createMemoryHistory } from 'history';
+import { defaultFormLease, defaultFormLeaseTerm, defaultLease } from 'interfaces';
+import { noop } from 'lodash';
+import { mockLookups } from 'mocks/mockLookups';
+import { lookupCodesSlice } from 'store/slices/lookupCodes';
+import { fillInput, renderAsync, RenderOptions, waitFor } from 'utils/test-utils';
+
+import { IPaymentsContainerProps, PaymentsContainer } from './PaymentsContainer';
+
+jest.mock('@react-keycloak/web');
+
+const history = createMemoryHistory();
+const mockAxios = new MockAdapter(axios);
+const storeState = {
+  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
+};
+const setLease = jest.fn();
+
+describe('PaymentsContainer component', () => {
+  const setup = async (
+    renderOptions: RenderOptions &
+      Partial<IPaymentsContainerProps> & {
+        initialValues?: any;
+      } = {},
+  ) => {
+    // render component under test
+    const component = await renderAsync(
+      <LeaseStateContext.Provider
+        value={{ lease: { ...defaultLease, ...renderOptions.initialValues, id: 1 }, setLease }}
+      >
+        <Formik initialValues={renderOptions.initialValues ?? {}} onSubmit={noop}>
+          <PaymentsContainer />
+        </Formik>
+      </LeaseStateContext.Provider>,
+      {
+        ...renderOptions,
+        history,
+        store: storeState,
+      },
+    );
+
+    return {
+      component,
+    };
+  };
+
+  beforeEach(() => {
+    mockAxios.resetHistory();
+  });
+  it('renders as expected', async () => {
+    const { component } = await setup({});
+
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+  it('renders with data as expected', async () => {
+    const { component } = await setup({
+      claims: [Claims.LEASE_EDIT],
+      initialValues: { ...defaultFormLease, terms: [defaultFormLeaseTerm] },
+    });
+
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('makes a post request when adding a new term', async () => {
+    const {
+      component: { getAllByText, getByText },
+    } = await setup({});
+    mockAxios.onPost().reply(200, { id: 1 });
+
+    const addButton = getAllByText('Add a Term')[0];
```

Comment:
nit: are you finding multiple "Add a Term" buttons here? couldn't we use `getByText('...')` instead?

Reply:
There is an add a term button and a add a term modal unfortunately.